Solutions/Nasuni/Analytic Rules/RansomwareAttackDetected.yaml

```diff
@@ -0,0 +1,50 @@
+id: 6c8770fb-c854-403e-a64d-0293ba344d5f
+name: Ransomware Attack Detected
+description: 'Identifies ransomware attacks detected by the Ransomware Protection service running on a Nasuni Edge Appliance.'
+kind: Scheduled
+severity: High
+requiredDataConnectors:
+  - connectorID: NasuniEdgeAppliance
+    datatypes: 
+      - Syslog
+queryFrequency: 5m
+queryPeriod: 5m
+triggerOperator: gt
+triggerThreshold: 0
+status: Available
+tactics:
+  - Impact
+relevantTechniques:
+  - T1486
+query: |-
+  Syslog
+  | project TimeGenerated, Computer, SyslogMessage
+  | where SyslogMessage has "The Filer has detected a new ransomware attack"
+  | extend pattern = substring(SyslogMessage, indexof(SyslogMessage, "(")+1, indexof(SyslogMessage, ")") - indexof(SyslogMessage,"(")-1)
+  | extend volume_name = substring(SyslogMessage, indexof(SyslogMessage, "volume")+7, indexof(SyslogMessage,". Visit") - (indexof(SyslogMessage, "volume")+7))
+  | sort by TimeGenerated desc
```

Comment:
where ever possible, please summarize result at the end e.g.
summarize EventCount=count() by volume_name, pattern , EventStartTime=min(TimeGenerated), EventEndTime=max(TimeGenerated)

Reply:
I don't think that summarization is appropriate here because this query is expected to return a single row. Please let me know if I am missing something.